draft-pda-protocol.md

```diff
@@ -146,13 +146,164 @@ document.
+protocol* is to compute `y = F(x_1, ..., x_n)` for some aggregation
+function `F` while revealing nothing else about the measurements.
+
+This protocol is extensible and allows for the addition of new
+cryptographic schemes that compute new functions. The current
+version supports two schemes:
+
+* Prio {{CB17}}, which allows for aggregate statistics such as
+  sum, mean, histograms, etc. over a single value.
+
+* Heavy Hitters {{BBCp21}}, which allows for finding the most
+  common strings among a collection of clients (e.g., the
+  URL of their home page) as well as counting the number of
+  clients that hold a given string.
+
+This protocol is designed to work with schemes that use secret
+sharing. Rather than send its input in the clear, each client splits
+its measurements into a sequence of *shares* and sends a share to each
+of the aggregators. This provides two important properties:
+
+* It's impossible to deduce the measurement without knowing *all* of the shares.
+
+* It allows the aggregators to compute the final output by first
+   aggregating up their measurements shares locally, then combining
+   the results to obtain the final output.
+
+## System Architecture
+
+The overall system architecture is shown in {{pa-topology}}.
+
+~~~~
+                    +------------+
+                    |            |
++--------+          |   Helper   |
+|        |          |            |
+| Client +----+     +-----^------+
+|        |    |           |
++--------+    |           |
+              |           |
++--------+    |     +-----v------+         +-----------+
+|        |    +----->            |         |           |
+| Client +---------->   Leader   +---------> Collector |
+|        |    +----->            |         |           |
++--------+    |     +-----^------+         +-----------+
+              |           |
++--------+    |           |
+|        |    |           |
+| Client +----+     +-----V------+
+|        |          |            |
++--------+          |   Helper   |
+                    |            |
+                    +------------+
+~~~~
+{: #pa-topology title="System Architecture"}
+
+
+The main participants in the protocol are as follows:
+
+Collector:
+: The entity which wants to take the measurement and ultimately receives
+  the results. Any given measurement will have a single collector.
+
+Client(s):
+: The endpoints which directly take the measurement(s) and report it to the
+  PPM system. In order to provide reasonable levels of privacy, there
+  must be a large number of clients.
+
+Aggregator:
+: An endpoint which receives report shares. Each aggregator works with the
+  other aggregators to compute the final aggregate. This protocol defines
+  two types of aggregators: a Leader and a set of Helpers.
+
+Leader:
+: The leader is responsible for coordinating the protocol. It receives
+  the encrypted shares, distributes them to the helpers and orchestrates
+  the process of computing the final measurement.
```

Comment:
```suggestion
  the process of computing the final measurement with the Collector.
```

Reply:
I see what you want, but I think this is a bit confusing as written
"maybe as requested by the Collector"

Reply:
Yeah, that's better